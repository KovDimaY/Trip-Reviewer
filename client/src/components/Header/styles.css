.header-container {
  padding: 5px;
  background: #00BCD4;
  position: relative;
  border-bottom: 6px solid #0495a8;
}

.header-container .logo {
  text-align: center;
  font-size: 30px;
  font-family: Breathe;
  letter-spacing: 3px;
  color: #ffffff;
  display: block;
}

.header-container .open-nav {
  position: absolute;
  left: 0px;
}