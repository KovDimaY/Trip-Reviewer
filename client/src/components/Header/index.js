import React, { Component } from 'react';
import FontAwesome from 'react-fontawesome';
import { Link } from 'react-router-dom';

import Nav from '../Sidenav/index';

import './styles.css';

class Header extends Component {
    state = {
      showNav: false,
    }

    handleHideNav = () => {
      this.setState({ showNav: false });
    }

    handleOpenNav = () => {
      this.setState({ showNav: true });
    }

    render() {
      return (
        <header className="header-container">
          <div className="open-nav">
            <FontAwesome
              name="bars"
              onClick={this.handleOpenNav}
              style={{
                color: '#ffffff',
                padding: '10px',
                cursor: 'pointer',
              }}
            />
          </div>
          <Nav
            showNav={this.state.showNav}
            onHideNav={this.handleHideNav}
          />

          <Link to="/" className="logo">
                    Travel Stories
          </Link>
        </header>
      );
    }
}

export default Header;
