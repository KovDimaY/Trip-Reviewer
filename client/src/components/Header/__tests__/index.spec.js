import React from 'react';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
 
import Header from './../../Header';

jest.mock('./../../Sidenav/index', () => 'Nav');
jest.mock('react-router-dom', () => ({ Link: 'Link' }));

const mockComponent = props => {
    return (
        <Header {...props} />
    );
};
 
describe('<Header />', () => {
    it('should render component', () => {
        const tree = create(mockComponent()).toJSON();
        
        expect(tree).toMatchSnapshot();
    });

    it('should change showNav state to false when handleHideNav is triggered', () => {
        const instance = mount(mockComponent()).instance();
    
        instance.handleHideNav();
    
        expect(instance.state.showNav).toEqual(false);
    });

    it('should change showNav state to true when handleOpenNav is triggered', () => {
        const instance = mount(mockComponent()).instance();
    
        instance.handleOpenNav();
    
        expect(instance.state.showNav).toEqual(true);
    });
});