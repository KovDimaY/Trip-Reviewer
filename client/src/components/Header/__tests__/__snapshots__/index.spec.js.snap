// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render component 1`] = `
<header
  className="header-container"
>
  <div
    className="open-nav"
  >
    <span
      aria-hidden={true}
      className="fa fa-bars"
      onClick={[Function]}
      style={
        Object {
          "color": "#ffffff",
          "cursor": "pointer",
          "padding": "10px",
        }
      }
    />
  </div>
  <Nav
    onHideNav={[Function]}
    showNav={false}
  />
  <Link
    className="logo"
    to="/"
  >
    Travel Stories
  </Link>
</header>
`;
